### PR TITLE
Notify the API caller which cluster upgrades are restricted by kubelet versions 

### LINKED ISSUE #2541: Notify the API caller which cluster upgrades are restricted by kubelet versions
**User Story**
When a user of the Kubermatic API is calling the `.../upgrades` endpoint to get the list of available upgrades, it needs to know which upgrades are not possible at the time due to incompatible kubeletes running in the cluster.

**Acceptance criteria**
 - [x] The v1 API structure `MasterVersion` has an additional boolean field `restrictedByKubeletVersion`
 - [x] `getClusterUpgrades` fetches all Machines and MachineDeployments within the cluster. When responding with the list of master versions, it marks `restrictedByKubeletVersion`=`true` for all versions that are deemed to by incompatible with existing Machines/MachineDeployments (using #2539)
 - [x] `TestGetClusterUpgradesV1` contains a scenario that verifies `restrictedByKubeletVersion`=`true`  for incompatible master-kubelet pairs

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**: Notifies the API caller which cluster upgrades are restricted by kubelet versions .

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Fixes #2541.

**Special notes for your reviewer**: n/a

**Documentation**: n/a

**Release note**:

```release-note
Mark cluster upgrades as restricted if kubelet version is incompatible.
```

### CODE REVIEW

File: api/pkg/handler/upgrade_test.go [hunk 1]
```diff
@@ -80,6 +86,51 @@ func TestGetClusterUpgradesV1(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "upgrade available but restricted by kubelet versions",
+			cluster: &kubermaticv1.Cluster{
```
Comment (p0lyn0mial) [inline api/pkg/handler/upgrade_test.go]: could you use `test.GenCluster` function ? for example:
```
cluster: func() *kubermaticv1.Cluster {
c := test.GenCluster(...)
// modify c
return c
}
```

**Comment by p0lyn0mial** [comment]
one `nit` and this lgtm

File: api/pkg/handler/upgrade_test.go (hunk 1 above)
Comment (maciaszczykm) [reply api/pkg/handler/upgrade_test.go]: Of course 👍

**Comment by maciaszczykm** [comment]
@p0lyn0mial PTAL

**Comment by p0lyn0mial** [comment]
@maciaszczykm thx

/lgtm
/approve

**Pull request merged**